src/components/component.js

```diff
@@ -0,0 +1,35 @@
+import { ComponentService } from '../services/component.service';
+
+export class Component {
+  constructor() {
+    this.hash = new ComponentService().generateHash();
```

Comment:
predefine componentService, and then use its methods

Reply:
done

Reply:
done